src/Symfony/Component/DependencyInjection/Compiler/ResolveBindingsPass.php

```diff
@@ -88,8 +89,14 @@ protected function processValue($value, $isRoot = false)
 
         $calls = $value->getMethodCalls();
 
-        if ($constructor = $this->getConstructor($value, false)) {
-            $calls[] = array($constructor, $value->getArguments());
+        try {
+            if ($constructor = $this->getConstructor($value, false)) {
+                $calls[] = array($constructor, $value->getArguments());
+            }
+        } catch (RuntimeException $e) {
+            $this->container->getDefinition($this->currentId)->addError($e->getMessage());
```

Comment:
should be `$value->addError()`, no ?

Reply:
this could be an inline definition, but we want the main definition to hold the error